lib/matplotlib/widgets.py

```diff
@@ -2007,113 +2045,231 @@ def new_axes(self, ax):
+            dv = v - vpress
+            vmin += dv
+            vmax += dv
 
+        # resize an existing shape
+        elif self.active_handle and self.active_handle != 'C':
+            vmin, vmax = self._extents_on_press
+            if self.active_handle == 'min':
+                vmin = v
+            else:
+                vmax = v
+        # new shape
+        else:
+            vmin, vmax = vpress, v
             if vmin > vmax:
                 vmin, vmax = vmax, vmin
+
+        self.extents = vmin, vmax
+
+        if self.onmove_callback is not None:
             self.onmove_callback(vmin, vmax)
 
-        self.update()
         return False
 
-    def _set_span_xy(self, event):
-        """Set the span coordinates."""
-        x, y = self._get_data(event)
-        if x is None:
-            return
+    def _draw_shape(self, vmin, vmax):
+        if vmin > vmax:
+            vmin, vmax = vmax, vmin
+        if self.direction == 'horizontal':
+            self.rect.set_x(vmin)
+            self.rect.set_width(vmax - vmin)
+        else:
+            self.rect.set_y(vmin)
+            self.rect.set_height(vmax - vmin)
+
+    def _set_active_handle(self, event):
+        """Set active handle based on the location of the mouse event."""
+        # Note: event.xdata/ydata in data coordinates, event.x/y in pixels
+        e_idx, e_dist = self._edge_handles.closest(event.x, event.y)
+
+        # Prioritise center handle over other handles
+        # Use 'C' to match the notation used in the RectangleSelector
+        if 'move' in self.state:
+            self.active_handle = 'C'
+        elif e_dist > self.maxdist:
+            # Not close to any handles
+            self.active_handle = None
+            if self.drag_from_anywhere and self._contains(event):
+                # Check if we've clicked inside the region
+                self.active_handle = 'C'
+                self._extents_on_press = self.extents
+            else:
+                self.active_handle = None
+                return
+        else:
+            # Closest to an edge handle
+            self.active_handle = self._edge_order[e_idx]
 
-        self.prev = x, y
+        # Save coordinates of rectangle at the start of handle movement.
+        self._extents_on_press = self.extents
+
+    def _contains(self, event):
+        """Return True if event is within the patch."""
+        return self.rect.contains(event, radius=0)[0]
+
+    @property
+    def vmin(self):
+        """Get the start span coordinate."""
         if self.direction == 'horizontal':
-            v = x
+            vmin = self.rect.get_x()
         else:
-            v = y
+            vmin = self.rect.get_y()
+        return vmin
+
+    @property
+    def vmax(self):
+        """Get the end span coordinate."""
+        if self.direction == 'horizontal':
+            vmax = self.vmin + self.rect.get_width()
+        else:
+            vmax = self.vmin + self.rect.get_height()
+        return vmax
```

Comment:
Can we use a name other than `vmin` and `vmax`? Those are very clearly associated with colormaps and i think I would be confused to see them here.